api/envoy/admin/v2alpha/config_dump.proto

```diff
@@ -70,9 +69,52 @@ message ListenersConfigDump {
     api.v2.Listener listener = 2;
 
     // The timestamp when the Listener was last updated.
+    // FIXME: reviewer we may eventually want last updated and last _successfully_ updated.
+    // Given similarity between last_updated (attempt or success) and
+    // last_update should this maybe be last_update_attempt so last_update
+    // matches?
     google.protobuf.Timestamp last_updated = 3;
   }
 
+  // Describes a dynamically loaded listener via the LDS API.
+  // FIXME rename? because otherwise v2->alpha is super confusing but this seems
+  // like the right name :-/
+  message DynamicListener {
+    // FIXME worth splitting name out for easy reference, duplicate cruft?
+    // The name or uniqe id of this listener, pulled from the DynamicListenerState config.
+    string name = 1;
+
+    // The listener state for any active listener by this name.
+    DynamicListenerState active_state = 2;
+
+    // The listener state for any warming listener by this name.
+    DynamicListenerState warming_state = 2;
+
+    // The listener state for any draining listener by this name.
+    DynamicListenerState draining_state = 3;
+
+    // FIXME alternately we could just have
+    // DynamicListenerState failure_state = 4;
+    // string error_details = 5;
+    // rather than the sub-message
+    //
+    // FIXME alternately alternately we could have shared UpdateErrorState with
+    // a Message error_state (which would be DynamicListenerState for listeners,
+    // DynamicClusterState for clusters, and error details for all.
+    // I like shared structures but hate "casting" so I'm scrapping reuse.
+    message ListenerUpdateErrorState {
+      // What this listener would have looked like if it had succeeded
+      // FIXME is it confusing to use last_updated to last_attempted_update?
```

Comment:
A bit, yes, but if documented it's probably fine, as the time it failed is actually useful IMO.